File: backend/app/models/meeting.py
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

MeetingStatus = Literal["queued", "transcribing", "transcribed", "summarizing", "done", "error"]
Language = Literal["en", "tr"]


class Meeting(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    title: str
    audio_url: str | None = None
    duration_seconds: int | None = None
    language: Language | None = None
    status: MeetingStatus
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime


class MeetingCreate(BaseModel):
    title: str | None = None


class MeetingUpdate(BaseModel):
    title: str | None = None


class MeetingStatusOut(BaseModel):
    status: MeetingStatus
    error_message: str | None = None

File: frontend/src/types/meeting.ts
export type MeetingStatus = 'queued' | 'transcribing' | 'transcribed' | 'summarizing' | 'done' | 'error';
export type ProcessingMode = 'summary' | 'transcript';
export type Language = 'en' | 'tr';

export interface Meeting {
  id: string;
  user_id: string;
  title: string;
  audio_url: string | null;
  duration_seconds: number | null;
  language: Language | null;
  status: MeetingStatus;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface MeetingCreate {
  title?: string;
}

export interface MeetingUpdate {
  title?: string;
}
